lib/facter/openssl_version.rb

```diff
@@ -4,7 +4,8 @@
   setcode do
     if Facter::Util::Resolution.which('openssl')
       openssl_version = Facter::Util::Resolution.exec('openssl version 2>&1')
-      matches = %r{^OpenSSL ([\w.-]+)(\s+FIPS)?( +)([\d.]+)( +)([\w.]+)( +)([\d.]+)}.match(openssl_version)
+      # OracleLinux did some uppercase-lowercase-extras
+      matches = %r{^OpenSSL ([\w.]+)[ -]*(fips|FIPS|dev)? +([\d.]+) +([\w.]+) +([\d.]+) *(\([\w:. ]+\))?}.match(openssl_version)
```

Comment:
I'm not convinced yet that such a complicated regex is still required.

```ruby
Facter::Util::Resolution.exec('openssl version 2>&1').split(' ')[1]
```

Reply:
Since the revision character has been included in the version for a while, that might be ok. However, doing so would miss the fips information. For example, `OpenSSL 1.1.1k  FIPS` and `OpenSSL 1.0.2k-fips`. As I said in my PR, `As an aside, it seems like we could be adding another fact based on FIPS mode, and possibly the library information for v3, but that is best left to another PR.`